### PR TITLE
Client Memory Work / LRU Cache Updates

### PR DESCRIPTION
**Pull request by holgerd77**
Some client related memory work.

### CODE REVIEW

File: packages/client/lib/service/ethereumservice.ts [hunk 1]
```diff
@@ -80,9 +98,35 @@ export class EthereumService extends Service {
...
+    this._statsInterval = setInterval(
+      // eslint-disable-next-line @typescript-eslint/await-thenable
+      await this.stats.bind(this),
+      this.STATS_INTERVAL
+    )
+
     return true
   }
 
+  stats() {
```
Comment (g11tech) [inline packages/client/lib/service/ethereumservice.ts:115]: yes this is a nice placement for logging and triggering shutdown :+1:

**Comment by holgerd77** [comment]
Can this be merged?

**Comment by holgerd77** [comment]
Will reuse this branch for some more memory work, have set back to "WIP".

File: packages/client/lib/service/ethereumservice.ts [hunk 2]
```diff
@@ -80,9 +98,35 @@ export class EthereumService extends Service {
...
+    if (this.v8Engine !== null) {
+      const { used_heap_size, heap_size_limit } = this.v8Engine.getHeapStatistics()
+
+      const heapUsed = Math.round(used_heap_size / 1000 / 1000) // MB
+      const percentage = Math.round((100 * used_heap_size) / heap_size_limit)
+      this.config.logger.info(`Memory stats usage=${heapUsed} MB percentage=${percentage}%`)
+
+      if (percentage >= this.MEMORY_SHUTDOWN_THRESHOLD && !this.config.shutdown) {
+        this.config.logger.error('EMERGENCY SHUTDOWN DUE TO HIGH MEMORY LOAD...')
```
Comment (acolytec3) [inline packages/client/lib/service/ethereumservice.ts:124]: This is a nit but this feels a little shouty.  Can we use normal casing and not all capitals?

**Review by acolytec3 — COMMENTED** [review]
This looks good so far as it goes.  Still need to figure out why the client is ignoring the shutdown signal in the `sim` test before we can merge though.

**Review by g11tech — DISMISSED** [review]
lgtm

File: packages/client/lib/service/ethereumservice.ts (hunk 2 above)
Comment (g11tech) [reply packages/client/lib/service/ethereumservice.ts:124]: i think the reason calls for being shouty lol

Comment (holgerd77) [reply packages/client/lib/service/ethereumservice.ts:124]: Also like it TBH

**Review by acolytec3 — APPROVED** [review]
LGTM!

**Review by acolytec3 — APPROVED** [review]
LGTM!

**Pull request merged**